buildkite/data_source_team.go

```diff
@@ -2,84 +2,153 @@ package buildkite
 
 import (
 	"context"
-	"errors"
 	"fmt"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func dataSourceTeam() *schema.Resource {
-	return &schema.Resource{
-		ReadContext: dataSourceTeamRead,
-		Schema: map[string]*schema.Schema{
-			"id": {
-				Computed: true,
-				Type:     schema.TypeString,
+type teamDatasourceModel struct {
+	ID                        types.String `tfsdk:"id"`
+	UUID                      types.String `tfsdk:"uuid"`
+	Slug                      types.String `tfsdk:"slug"`
+	Name                      types.String `tfsdk:"name"`
+	Privacy                   types.String `tfsdk:"privacy"`
+	Description               types.String `tfsdk:"description"`
+	IsDefaultTeam             types.Bool   `tfsdk:"default_team"`
+	DefaultMemberRole         types.String `tfsdk:"default_member_role"`
+	MembersCanCreatePipelines types.Bool   `tfsdk:"members_can_create_pipelines"`
+}
+
+type teamDatasource struct {
+	client *Client
+}
+
+func newTeamDatasource() datasource.DataSource {
+	return &teamDatasource{}
+}
+
+func (t *teamDatasource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	t.client = req.ProviderData.(*Client)
+}
+
+func (t *teamDatasource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_team"
+}
+
+func (t *teamDatasource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This datasource allows you to get a team from Buildkite.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the team.",
+				Optional:            true,
+				Computed:            true,
 			},
-			"slug": {
-				Required: true,
-				Type:     schema.TypeString,
+			"uuid": schema.StringAttribute{
+				MarkdownDescription: "The UUID of the team.",
+				Computed:            true,
 			},
-			"uuid": {
-				Computed: true,
-				Type:     schema.TypeString,
+			"slug": schema.StringAttribute{
```

Comment:
I reckon for all these slug, id, uuid attributes, to allow us to use any of them to find the team we should add validation that only one is supplied to use as well: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.10.0/stringvalidator#ExactlyOneOf

Reply:
Ha, knew there was a thing but tbh this has been such a pain to implement due to the Node thing too I forgot.

Reply:
Yeah I bet its pretty crappy to implement. But I think its useful for users to use this way at least 🙏